sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetScanBuilder.scala

```diff
@@ -80,8 +88,84 @@ case class ParquetScanBuilder(
   // All filters that can be converted to Parquet are pushed down.
   override def pushedFilters(): Array[Filter] = pushedParquetFilters
 
+  override def pushAggregation(aggregation: Aggregation): Boolean = {
+
+    def getStructFieldForCol(col: FieldReference): StructField = {
+      schema.fields(schema.fieldNames.toList.indexOf(col.fieldNames.head))
+    }
+
+    def isPartitionCol(col: FieldReference) = {
+      (readPartitionSchema().fields.map(PartitioningUtils
+        .getColName(_, sparkSession.sessionState.conf.caseSensitiveAnalysis))
+        .toSet.contains(col.fieldNames.head))
+    }
+
+    def checkMinMax(agg: AggregateFunc): Boolean = {
+      val (column, aggType) = agg match {
+        case max: Max => (max.column(), "max")
+        case min: Min => (min.column(), "min")
+        case _ => throw new IllegalArgumentException(s"Unexpected type of AggregateFunc")
+      }
+
+      if (column.fieldNames.length != 1 || isPartitionCol(column)) {
+        return false
+      }
```

Comment:
Add a comment for this case?

Reply:
added comment
removed `column.fieldNames.length != 1` because nested column is already blocked